### PR TITLE
Builtin: add builtin function "TIMESTAMPADD"

### PR DESCRIPTION
**Pull request by spongedu**
Add implementation for builtin function: 'TIMESTAMPADD'

### CODE REVIEW

File: expression/builtin_time.go [hunk 1]
```diff
@@ -1982,7 +1982,59 @@ type builtinTimestampAddSig struct {
...
+	if err != nil {
+		return d, errorOrWarning(err, b.ctx)
+	}
+	tm, err := date.Time.GoTime(time.Local)
+	if err != nil {
+		return d, errors.Trace(err)
+	}
+	var tb time.Time
+	fsp := types.DefaultFsp
+	switch unit {
```
Comment (shenli) [inline expression/builtin_time.go]: Maybe you need to convert unit to upper case.

File: expression/builtin_time.go [hunk 2]
```diff
@@ -1982,7 +1982,59 @@ type builtinTimestampAddSig struct {
...
+	sc := b.ctx.GetSessionVars().StmtCtx
+	date, err := convertDatumToTime(sc, args[2])
+	if err != nil {
+		return d, errorOrWarning(err, b.ctx)
+	}
+	tm, err := date.Time.GoTime(time.Local)
+	if err != nil {
+		return d, errors.Trace(err)
+	}
+	var tb time.Time
```
Comment (coocood) [inline expression/builtin_time.go]: What name it `tb`?

**Comment by tiancaiamao** [comment]
LGTM

**Comment by ngaut** [comment]
LGTM

File: expression/builtin_time.go (hunk 1 above)
Comment (spongedu) [inline expression/builtin_time.go]: Parser have done this.

Comment (shenli) [inline expression/builtin_time.go]: ok

File: expression/builtin_time.go (hunk 2 above)
Comment (spongedu) [inline expression/builtin_time.go]: Short for 'time-built'. Maybe it's not very proper.... any suggestions?

Comment (coocood) [inline expression/builtin_time.go]: ok

**Pull request merged**